app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+
+  get '/users/:slug' do
+    @user= User.find_by_slug(params[:slug])
+    erb :'users/show_user'
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post "/signup" do
+    user = User.new(username: params[:username], email: params[:email], password: params[:password])
```

Comment:
I would check to make sure a user is logged_in? before you allow them to make a user. At the moment anyone could force create more users.